cli/cmd/cache.go

```diff
@@ -132,7 +132,7 @@ func (c *cliState) EraseCachedToken() error {
 }
 
 func (c *cliState) ReadCachedToken() {
-	if c.noCache {
+	if c.noCache || c.Cache == nil {
```

Comment:
I feel you catch some issues plugin this in, I am now against these additions but I want to understand why we need to check for the cache here and also, initialize the api somewhere else that is not the `root.go`?

Reply:
Prior to this we could send Honeycomb events without the client being initialized.

In the persistent prerun we always sent a Honeycomb event, say for example `lacework help`.
We then got an event in honeycomb with the cmd,version but no account/subaccount or any client data.

Now we cannot do that. We need Lacework auth to send a honeycomb event. This also means those honeycomb events with no client data will not be sent.